### PR TITLE
soc: arm: silabs_exx32: add support for sys_poweroff

### PR DESCRIPTION
**Pull request by gmarull**
Implement the sys_poweroff() hooks based on existing code for SOFT_OFF. Note that the Kconfig structure looks a bit messy, so I've done my best to understand it.

### CODE REVIEW

**Review by fkokosinski — CHANGES_REQUESTED** [review]

File: soc/arm/silabs_exx32/common/poweroff.c [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#include <sl_power_manager.h>
+#endif
+
+void z_sys_poweroff(void)
+{
+#if defined(CONFIG_SOC_GECKO_PM_BACKEND_EMU)
+	EMU_EnterEM4();
+#elif defined(CONFIG_SOC_GECKO_PM_BACKEND_PMGR)
+	sl_power_manager_add_em_requirement(SL_POWER_MANAGER_EM4);
+	sl_power_manager_sleep();
```
[the remark this answers is not in the document]
Comment (gmarull) [reply soc/arm/silabs_exx32/common/poweroff.c:21]: Are these dependencies specified somewhere in Kconfig? This looks wrong to me. This code is not using neither PM API nor counter API. Also, CONFIG_POWEROFF is independent of the PM subsys.

[the remark this answers is not in the document]
Comment (msierszulski) [reply soc/arm/silabs_exx32/common/poweroff.c:21]: Dependencies of `PM` can be seen here:

https://github.com/zephyrproject-rtos/zephyr/blob/40d85ebcc535e1f337ff52ef90f599b0101934e4/soc/arm/silabs_exx32/efr32bg22/Kconfig.defconfig.series#L18-L20 or
 https://github.com/zephyrproject-rtos/zephyr/blob/40d85ebcc535e1f337ff52ef90f599b0101934e4/soc/arm/silabs_exx32/efr32bg27/Kconfig.defconfig.series#L18-L19

We would have to specify the `COUNTER` dependency in the same way for `CONFIG_POWEROFF`.

[the remark this answers is not in the document]
Comment (msierszulski) [reply soc/arm/silabs_exx32/common/poweroff.c:21]: Actually, one `if` is missing in `efr32bg22`.

It should be done as it is in:
https://github.com/zephyrproject-rtos/zephyr/blob/b6095a878c7e9b330dc4a9d91a6a107c544fb47a/soc/arm/silabs_exx32/efr32mg24/Kconfig.defconfig.series#L19-L22

[the remark this answers is not in the document]
Comment (gmarull) [reply soc/arm/silabs_exx32/common/poweroff.c:21]: I still feel what is being done here is wrong. `PM` should not select dependencies it doesn't use. If your SoC requires these subsystems, it should be like

```
config COUNTER
    default y if PM
```

**Pull request closed without merging**